zerver/lib/message.py

```diff
@@ -332,14 +333,15 @@ def render_markdown(message, content, realm_id=None, realm_alert_words=None, mes
     rendered_content = bugdown.convert(content, realm_id=realm_id, message=message,
                                        possible_words=possible_words)
 
-    message.user_ids_with_alert_words = set()
+    if message is not None:
+        message.user_ids_with_alert_words = set()
 
-    if realm_alert_words is not None:
-        for user_id, words in realm_alert_words.items():
-            if user_id in message_user_ids:
-                if set(words).intersection(message.alert_words):
-                    message.user_ids_with_alert_words.add(user_id)
+        if realm_alert_words is not None:
+            for user_id, words in realm_alert_words.items():
+                if user_id in message_user_ids:
+                    if set(words).intersection(message.alert_words):
+                        message.user_ids_with_alert_words.add(user_id)
 
-    message.is_me_message = Message.is_status_message(content, rendered_content)
+        message.is_me_message = Message.is_status_message(content, rendered_content)
```

Comment:
This refactoring to support calling this function without a message object should probably be split out as its own commit (earlier in the branch than the main emoji search feature).  This is part of our general principle of separating (and quickly merging) splittable refactorings independently from new features to make it easy to confirm changes are correct.

Check out the Zulip version control guidelines for more details: https://zulip.readthedocs.io/en/latest/version-control.html